client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -2489,45 +2489,61 @@ Undocumented.prototype.startMigration = function( sourceSiteId, targetSiteId ) {
 Undocumented.prototype.getAtomicSiteMediaViaProxy = function(
 	siteIdOrSlug,
 	mediaPath,
-	{ query = '', maxSize },
-	fn
+	{ query = '', maxSize }
 ) {
 	const safeQuery = query.replace( /^\?/, '' );
 	const params = {
 		path: `/sites/${ siteIdOrSlug }/atomic-auth-proxy/file?path=${ mediaPath }&${ safeQuery }`,
 		apiNamespace: 'wpcom/v2',
 	};
 
-	const fetchMedia = () => this.wpcom.req.get( { ...params, responseType: 'blob' }, fn );
-
-	if ( ! maxSize ) {
-		return fetchMedia();
-	}
+	return new Promise( ( resolve, _reject ) => {
+		const fetchMedia = () =>
+			this.wpcom.req.get( { ...params, responseType: 'blob' }, ( error, data ) => {
+				if ( error || ! ( data instanceof Blob ) ) {
+					_reject( error );
```

Comment:
The non-error / invalid response type case won't be passed through the error chain like this.  Should we pass the invalid response back in these instances? Something like:

```suggestion
				const success = ! error && data instanceof Blob;
				if ( ! success ) {
					_reject( error || { error: 'invalidResponse', data } );
```